File: packages/core/src/ids.ts
import { randomBytes } from "node:crypto";

const ALPHABET = "0123456789abcdefghjkmnpqrstvwxyz"; // Crockford-ish, no ambiguous chars

/** Short, sortable-enough, prefixed id: `inv_7k2p9x...` */
export function id(prefix: string, len = 16): string {
  const bytes = randomBytes(len);
  let out = "";
  for (const b of bytes) out += ALPHABET[b % ALPHABET.length];
  return `${prefix}_${out}`;
}

File: server/internal/ids/ids.go
// Package ids makes short, prefixed, URL-safe identifiers.
package ids

import (
	"crypto/rand"
	"fmt"
)

// Crockford-ish: no vowels to form words by accident, no characters that look
// alike when read aloud or copied by hand.
const alphabet = "0123456789abcdefghjkmnpqrstvwxyz"

// New returns something like "inv_7k2p9xq3mtv4bn8d".
func New(prefix string) string {
	b := make([]byte, 16)
	if _, err := rand.Read(b); err != nil {
		panic(fmt.Sprintf("crypto/rand failed: %v", err))
	}
	out := make([]byte, len(b))
	for i, v := range b {
		out[i] = alphabet[int(v)%len(alphabet)]
	}
	return prefix + "_" + string(out)
}
